synapse/rest/client/v2_alpha/register.py

```diff
@@ -743,6 +729,66 @@ def _do_guest_registration(self, params, address=None):
         }))
 
 
+def cap(s):
+    """Capitalise words in a string, including examples such as
+    'John-Doe'"""
+    if not s:
+        return s
+
+    # Split phrase by spaces and hyphens
+    # We will end up with a list of lists, where strings in each sublist must be
+    # joined by hyphens
+    # e.g 'jack-phill person' -> [['jack', 'phill'], ['person']]
+    s = [x.split("-") for x in s.split()]
+
+    # Capitalise each word in each sublist
+    # [['jack', 'phill'], ['person']] -> [['Jack', 'Phill'], ['Person']]
+    for i in range(len(s)):
+        inner_list = s[i]
+        for j in range(len(inner_list)):
+            s[i][j] = s[i][j].capitalize()
```

Comment:
I find it hard to follow the subscripts here tbh (and you probably would have wanted to use `enumerate()`). I'd be tempted to change this to something like:

```python
space_delimited = []  # type: List[str]
for space_part in name.split():
    hyphen_delimited = []  # type: List[str]
    for hyphen_part in space_part.split("-"):
        hyphen_delimited.append(hyphen_part.capitalize())

    space_delimited.append("-".join(hyphen_delimited))

capatilized_name = " ".join(space_delimited)
```

or even

```python
# Split the name by whitespace then hyphens, capitalizing each part then joining it
# back together.
capatilized_name = " ".join(
    "-".join(part.capitalize() for part in space_part.split("-"))
    for space_part in name.split()
)
```